rust/src/tests/errors/efault.rs

```diff
@@ -0,0 +1,108 @@
+/// Create a test case which asserts that the sycall
+/// returns EFAULT if the path argument points
+/// outside the process's allocated address space.
+/// It takes a function with the path pointer as argument.
+///
+/// ```ignore
+/// efault_path_test_case!(|ptr| nix::libc::mkdir(ptr, 0o755))
+/// ````
+macro_rules! efault_path_test_case {
+    ($syscall: ident, $fn: expr) => {
+        crate::test_case! {
+            #[doc = concat!(stringify!($syscall),
+            " returns EFAULT if the path argument points",
+            " outside the process's allocated address space"
+            )]
+            efault_path
+        }
+        fn efault_path(_: &mut crate::runner::context::TestContext) {
+            let f = |ptr| unsafe { $fn(ptr) };
+
+            let null_ptr = std::ptr::null();
+            // TODO: This theorically could be a valid pointer, but
```

Comment:
I think it's fine. Even if this is a valid u8 pointer, it can't possibly be a valid pointer for anything larger.